Core/RetroAchievements.cpp

```diff
@@ -51,8 +58,59 @@
 #include "Core/ELF/ParamSFO.h"
 #include "Core/System.h"
 #include "Core/FileSystems/MetaFileSystem.h"
+#include "Core/FileSystems/ISOFileSystem.h"
 #include "Core/RetroAchievements.h"
 
+static bool HashISOFile(ISOFileSystem *fs, const std::string filename, md5_context *md5) {
+	int handle = fs->OpenFile(filename, FILEACCESS_READ);
+	if (handle < 0) {
+		return false;
+	}
+
+	uint32_t sz = fs->SeekFile(handle, 0, FILEMOVE_END);
+	fs->SeekFile(handle, 0, FILEMOVE_BEGIN);
+	if (!sz) {
+		return false;
+	}
+
+	std::unique_ptr<uint8_t[]> buffer(new uint8_t[sz]);
```

Comment:
How about
`auto buffer = std::make_unique<uint8_t[]>(sz);`
?

Reply:
I'll sneak this into an upcoming change, thanks.